### PR TITLE
[telemetry] fix: reject duplicate CUDA masks

### PR DESCRIPTION
**Pull request by Wangmerlyn**
## Summary
- treat duplicate CUDA visibility masks as unavailable utilization instead of aliasing visible ranks to one physical GPU
- normalize numeric CUDA_VISIBLE_DEVICES tokens so aliases like `0,00` are rejected before NVML lookup
- add fake-NVML no-GPU tests for exact numeric duplicates, equivalent numeric aliases, and duplicate UUID tokens
- update AGENTS.md, README, architecture/getting-started/guides/reference docs, and add the implementation plan

## Verification
- `PYTHONPATH=$PWD/src pytest tests/utilities/test_gpu_monitor.py -q` -> 18 passed
- `PYTHONPATH=$PWD/src pytest tests/utilities/test_gpu_monitor.py tests/utilities/test_gpu_info.py tests/utilities/test_platform_manager.py -q` -> 42 passed, 1 skipped
- `PYTHONPATH=$PWD/src pytest tests -q` -> 266 passed, 11 skipped
- `PYTHONPATH=$PWD/src mkdocs build` -> passed with existing Material/nav warnings
- `pre-commit run --all-files` -> passed
- `git diff --check && git diff --cached --check` -> clean

## Local review
- Local subagent review found and verified the `0,00` numeric alias gap. Follow-up review reported the blocker and minor test/plan findings resolved, with no remaining Critical or Important blockers.

### CODE REVIEW

File: src/keep_gpu/utilities/gpu_monitor.py [hunk 1]
```diff
@@ -80,6 +80,12 @@ def _get_handle_for_visible_index(self, index: int):
             return self._nvml.nvmlDeviceGetHandleByIndex(index)
 
         tokens = [token.strip() for token in cuda_visible_devices.split(",")]
+        token_keys = [
+            ("index", int(token)) if token.isdigit() else ("token", token)
+            for token in tokens
+        ]
+        if len(set(token_keys)) != len(token_keys):
+            return None
```
[the remark this answers is not in the document]
Comment (Wangmerlyn) [reply src/keep_gpu/utilities/gpu_monitor.py:89]: Fixed in a44c51f. Duplicate-key detection now ignores empty tokens so existing malformed-token lookup behavior is preserved, and lowercases non-digit token keys so UUID aliases like `GPU-abc123,gpu-abc123` are treated as duplicates before NVML lookup. Added focused regressions for both cases: `test_monitor_returns_none_for_case_insensitive_duplicate_uuid_mask` and `test_monitor_preserves_valid_rank_before_repeated_empty_tokens`. Fresh local verification: `pytest tests/utilities/test_gpu_monitor.py -q` -> 20 passed; targeted utilities -> 44 passed, 1 skipped; full tests -> 268 passed, 11 skipped; mkdocs and pre-commit passed.
